components/eamxx/src/control/intensive_observation_period.cpp

```diff
@@ -189,137 +168,82 @@ read_fields_from_file_for_iop (const std::string& file_name,
-  dp_ic_reader.read_variables();
-  dp_ic_reader.finalize();
+  AtmosphereInput file_reader(file_name,io_grid,io_fields);
+  file_reader.read_variables();
+  file_reader.finalize();
 
   // For each field, broadcast data from closest lat/lon column to all processors
-  // and copy data into each field's column.
-
-  // Find the maximum size of transfer data over fields.
-  int max_data_size = 0;
-  for (auto& f : fields) {
-    const auto fl = f.get_header().get_identifier().get_layout();
-    if (fl.size()/fl.dim(0) > max_data_size) max_data_size = fl.size()/fl.dim(0);
-  }
-
-  Real transfer_data[max_data_size];
-  for (size_t i=0; i<fields.size(); ++i) {
-    const auto f = fields[i];
-    const auto fl = f.get_header().get_identifier().get_layout();
-    const auto field_rank = fl.rank();
-    const auto fname = field_names_eamxx[i];
-
-    // If I am the rank with the minimum lat/lon distance,
-    // store data of values to broadcast.
-    const auto my_rank = m_comm.rank();
-    const auto col_mpi_rank = m_lat_lon_info[grid_name].mpi_rank_of_closest_column;
-    if (my_rank == col_mpi_rank) {
-      const auto col_idx = m_lat_lon_info[grid_name].local_column_index_of_closest_column;
-      switch (field_rank) {
-        case 1: {
-          auto src_view = f.get_view<Real*, Host>();
-          transfer_data[0] = src_view(col_idx);
-          break;
-        }
-        case 2: {
-          auto src_view = f.get_view<Real**, Host>();
-          for (auto i1=0; i1<fl.dim(1); ++i1) {
-            transfer_data[i1] = src_view(col_idx, i1);
-          }
-          break;
-        }
-        case 3: {
-          auto src_view = f.get_view<Real***, Host>();
-          for (auto i1=0; i1<fl.dim(1); ++i1) {
-            for (auto i2=0; i2<fl.dim(2); ++i2) {
-              transfer_data[i2+i1*fl.dim(2)] = src_view(col_idx, i1, i2);
-          }}
-          break;
-        }
-        default:
-          // Currently only up to rank 3 inputs needed from IC file.
-          // Additional ranks can easily be added if needed.
-          EKAT_ERROR_MSG ("Error! Unexpected field rank (" + std::to_string(field_rank) + ") when transferring "
-                          "IOP ICs for field "+fname+". If needed, this can be implemented by adding another case.\n");
+  // and copy data into each field's column in the field manager.
+  for (size_t i=0; i<io_fields.size(); ++i) {
+    const auto& io_field = io_fields[i];
+    const auto& fname = field_names_eamxx[i];
+    auto& fm_field = field_mgr->get_field(fname);
+
+    // Create a temporary field to store the data from the
+    // single column of the closest lat/lon pair
+    const auto io_fid = io_field.get_header().get_identifier();
+    auto dims = io_fid.get_layout().dims();
+    dims[0] = 1;
+    FieldLayout col_data_fl(io_fid.get_layout().tags(), dims);
+    FieldIdentifier col_data_fid("col_data", col_data_fl, dummy_units, "");
+    Field col_data(col_data_fid);
```

Comment:
This is fine, but fyi, rank-0 fields are supported in EAMxx (if you tried and they didn't, let me know!). So you could have done
```
auto col_lt = io_fid.get_layout().strip_dim(FieldTag::Column);
```
This can potentially give you a rank 0 field, but those should work. Then, later on, you don't need checks on `col_data.rank()`, you should be able to simply do, e.g.
```
col_data.deep_copy(io_field.subfield(0,col_indx_with_data);
```
and
```
fm_field.subfield(0,icol).deep_copy<Host>(col_data);
```

Reply:
I think then the field.deep_copy()is what complained. Let me check, if that is the case I think it should be implemented and I'll switch designs.